packages/cardano-services/src/WsServer/server.ts

```diff
@@ -0,0 +1,319 @@
+// cSpell:ignore njson
+
+import {
+  Cardano,
+  CardanoNode,
+  NetworkInfoResponses,
+  Seconds,
+  WSMessage,
+  createSlotEpochInfoCalc
+} from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { NJSON } from 'next-json';
+import { Notification, Pool } from 'pg';
+import { Server, createServer } from 'http';
+import { WebSocket, WebSocketServer } from 'ws';
+import { getLovelaceSupply, getProtocolParameters, getStake } from './requests';
+import { initDB } from './db';
+import { v4 } from 'uuid';
+
+export interface WsServerConfiguration {
+  /** The cache time to live in seconds. */
+  dbCacheTtl: number;
+
+  /** The heartbeat timeout in seconds. */
+  heartbeatTimeout?: number;
+
+  /** The port to listen. */
+  port: number;
+}
+
+export interface WsServerDependencies {
+  /** The `CardanoNode` object. */
+  cardanoNode: CardanoNode;
```

Comment:
Wouldn't this be be a more suitable dependency, since it already emits some data that we would otherwise have to query? It might be missing some properties that we'd need to add though.
```suggestion
  cardanoNode: ObservableCardanoNode;
```

Reply:
I didn't checked it; ATM `CardanoNode` does its job, maybe we could decouple this as a separated improvement task?

Reply:
Sure - not a required change for a POC. Do we have an epic/story for a production-ready ws provider yet where we could log this?

Reply:
Not yes, ATM it belongs to the BEv2 phase 1 LW-10676

Reply:
LW-10840